subsys/jwt/jwt.c

```diff
@@ -154,23 +159,17 @@ static struct json_obj_descr jwt_payload_desc[] = {
  */
 static void jwt_add_header(struct jwt_builder *builder)
 {
-	static const struct jwt_header head = {
-		.typ = "JWT",
-#ifdef CONFIG_JWT_SIGN_RSA
-		.alg = "RS256",
-#endif
-#ifdef CONFIG_JWT_SIGN_ECDSA
-		.alg = "ES256",
-#endif
-	};
+	int jwt_header_len = ARRAY_SIZE(jwt_header);
 
-	int res = json_obj_encode(jwt_header_desc, ARRAY_SIZE(jwt_header_desc),
-				  &head, base64_append_bytes, builder);
-	if (res != 0) {
-		/* Log an error here. */
-		return;
+	strncpy(builder->buf, jwt_header, builder->len);
+	if (jwt_header_len <= builder->len) {
+		builder->buf += jwt_header_len - 1;
+		builder->len -= jwt_header_len - 1;
+	} else {
+		builder->overflowed = true;
```

Comment:
This should at least be safe, although in this branch the buf and len values won't point to the newly appended string. Realistically, there isn't really any hope for the JWT if the header doesn't even fit in the buffer, so as long as we have overflowed set, it should be fine.

Reply:
added correct return value in jwt_init_builder in case of potential buffer overflow